tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Logging/ILogger/DirectSubmission/LoggerFactoryIntegrationCommon.cs

```diff
@@ -0,0 +1,81 @@
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(LoggerFactoryIntegrationCommon<TLoggerFactory>));
+
+        // ReSharper disable once StaticMemberInGenericType
+        // Internal for testing
+        internal static readonly Type ProviderInterfaces;
+
+        static LoggerFactoryIntegrationCommon()
+        {
+            // The ILoggerProvider type is in a different assembly to the LoggerFactory, so go via the ILogger type
+            // returned by CreateLogger
+            var loggerFactoryType = typeof(TLoggerFactory);
+            var abstractionsAssembly = loggerFactoryType.GetMethod("CreateLogger")?.ReturnType.Assembly;
+            var iLoggerProviderType = abstractionsAssembly.GetType("Microsoft.Extensions.Logging.ILoggerProvider");
+            var iSupportExternalScopeType = abstractionsAssembly.GetType("Microsoft.Extensions.Logging.ISupportExternalScope");
+
+            if (iSupportExternalScopeType is null)
+            {
+                // ISupportExternalScope is only available in v2.1+
+                // We can just duck type ILoggerProvider directly
+                ProviderInterfaces = iLoggerProviderType;
+            }
+
+            // We need to implement both ILoggerProvider and ISupportExternalScope
+            // because LoggerFactory uses pattern matching to check if we implement the latter
+            // Duck Typing can currently only implement a single interface, so emit
+            // a new interface that implements both ILoggerProvider and ISupportExternalScope
+            // and duck cast to that
+            var thisAssembly = typeof(DirectSubmissionLoggerProvider).Assembly;
+            var assemblyName = new AssemblyName("DirectLogSubmissionILoggerFactoryAssembly")
+            {
+                Version = thisAssembly.GetName().Version
+            };
+
+            var assemblyBuilder = AssemblyBuilder.DefineDynamicAssembly(assemblyName, AssemblyBuilderAccess.Run);
+            var moduleBuilder = (ModuleBuilder)assemblyBuilder.DefineDynamicModule("MainModule");
+
+            var typeBuilder = moduleBuilder.DefineType(
+                "DirectSubmissionLoggerProviderProxy",
+                TypeAttributes.Interface | TypeAttributes.Public | TypeAttributes.Abstract,
+                parent: null,
+                interfaces: new[] { iLoggerProviderType, iSupportExternalScopeType });
+
+            ProviderInterfaces = typeBuilder.CreateTypeInfo().AsType();
+        }
+
+        internal static object AddDirectSubmissionLoggerProvider(TLoggerFactory loggerFactory)
+        {
+            var provider = new DirectSubmissionLoggerProvider();
+
+            return AddDirectSubmissionLoggerProvider(loggerFactory, provider);
+        }
+
+        // Internal for testing
+        internal static object AddDirectSubmissionLoggerProvider(TLoggerFactory loggerFactory, DirectSubmissionLoggerProvider provider)
+        {
+            // Need tto use the
```

Comment:
Not a big deal but what was the intent of this comment?

Reply:
Who knows, I wrote it 2 months ago 🤣

Reply:
removed in https://github.com/DataDog/dd-trace-dotnet/pull/1945/commits/140cb5ac79f702b3fd3fc45d176f8b3a851c32af